### PR TITLE
Feat/handle interupts

### PR DESCRIPTION
**Pull request by unparalleled-js**
### What I did

Related issue: #158 

### How I did it

Add this code to the top of `ape.__init__.py` (earliest point in execution).
```python
import signal

signal.signal(signal.SIGINT, lambda s, f: _sys.exit(130))
```

Note that it is best to put it here because it is a lengthy process to import the managers from ape, so you can easily interrupr the process at that point and it is not graceful.

Additionally, this violates `flake8` E402, which is definitely a good rule but I feel at this time that this is an edge case so I did a `flake8` per-file ignore for this rule in the `flake8` config.

### Checklist

- [ ] Passes all linting checks (pre-commit and CI jobs)
- [ ] New test cases have been added and are passing
- [ ] Documentation has been updated
- [ ] PR title follows [Conventional Commit](https://www.conventionalcommits.org/en/v1.0.0/) standard (will be automatically included in the changelog)

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -4,3 +4,6 @@ exclude =
 	venv*
 	docs
 	build
+per-file-ignores =
+    # Need signal handler before imports
+    src/ape/__init__.py: E402
```
Comment (fubuloubu) [inline setup.cfg:9]: why not `# noqa: E402` in the file?

File: src/ape/__init__.py [hunk 2]
```diff
@@ -1,3 +1,7 @@
+import signal
+
+signal.signal(signal.SIGINT, lambda s, f: _sys.exit(130))
```
Comment (fubuloubu) [inline src/ape/__init__.py:3]: Have to try this locally...

File: setup.cfg (hunk 1 above)
Comment (unparalleled-js) [reply setup.cfg:9]: I could put `# noqa: E402` after every import in the file, but that is effectively that same as this.

I couldn't figure out a way to ignore the entire file using a comment within the file, unless you know?

File: src/ape/__init__.py (hunk 2 above)
Comment (unparalleled-js) [reply src/ape/__init__.py:3]: I am wondering if this a better place to register this.
I generally don't put them here but more like in the `_cli.py` but this file actually takes a long time to load.

File: setup.cfg (hunk 1 above)
Comment (fubuloubu) [reply setup.cfg:9]: ah, I misunderstood. this is fine lol

I think there's a way to do it at the top of the file, but I don't remember how.

File: src/ape/__init__.py (hunk 2 above)
Comment (fubuloubu) [reply src/ape/__init__.py:3]: `_cli.py` might make more sense. that's the actual entry point for running this code I believe

File: setup.cfg (hunk 1 above)
Comment (fubuloubu) [reply setup.cfg:9]: Wonder if you could just get `isort` to ignore the ordering of the import :thinking:

**Review by fubuloubu — APPROVED** [review]
Tried it locally, looks great!

Did the following:
1. Run `ape`
2. Hit `ctrl+C` prior to loading
3. Observe lack of stack trace

File: src/ape/__init__.py (hunk 2 above)
Comment (unparalleled-js) [reply src/ape/__init__.py:3]: For some reason, lol, `__init__` happens first.
I put print statements at the top of both modules and run `ape`.  The `__init__.py` one ran first.  Is this unexpected? Kind of interesting.

File: setup.cfg (hunk 1 above)
Comment (unparalleled-js) [reply setup.cfg:9]: I think `isort` is actually okay with this. `flake8` i the one with the problem.

File: src/ape/__init__.py (hunk 2 above)
Comment (fubuloubu) [reply src/ape/__init__.py:3]: hmm, entry points must load the module first. weird.
